File: 1608. Special Array With X Elements Greater Than or Equal X.java
class Solution {
    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        for (int i = nums.length-1; i >= 0; i--) {
            int greater = nums.length-i;
            if (i > 0 && nums[i-1] < greater && nums[i] >= greater) {
                return greater;
            }
            if (i == 0 && nums[i] >= greater) {
                return greater;
            }
        }
        return -1;
    }
}

File: 1608. Special Array With X Elements Greater Than or Equal X.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        for i in range(len(nums)):
            if nums[i] >= len(nums)-i:
                if i == 0 or nums[i-1] < len(nums)-i:
                    return len(nums)-i
        return -1